doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+ArsOrch uses this to create the ARS SAI profile.
+
+4. Add ARS_INTERFACE Table
+The CONFIG_DB adds the ARS_INTERFACE table, causing ArsOrch to enable ARS on the corresponding ports.
+
+__Figure 5: Interface update flow__
+![](images/if_update.png)
+
+1. Link UP Notification
+The PortsOrch component detects a link state change and notifies ArsOrch when a port transitions to the "UP" state.
+
+2. Set Port Scaling Factor
+Based on the port's speed, PortsOrch determines the appropriate scaling factor.
+The scaling factor is then set in the SAI layer through the syncd component to adjust the port's behavior accordingly.
+
+__Figure 6: ARS Nexthop group table creation flow__
+![](images/ars_nhg_create.png)
+
+1. Add ARS_NEXTHOP_GROUP Table
+    * The CONFIG_DB adds the ARS_NEXTHOP_GROUP table, triggering the process.
+2. Create SAI ARS Object
+    * The ArsOrch component receives the update and creates the SAI ARS object.
+3. ArsOrch binds to events from RouteOrch related to the newly created NHG.
+4. Notification
+    * RouteOrch component will notify ArsOrch of NHG update.
+3. Nexthop group's SAI attributes handling:
+    * If nexthop group's ARS object ID, "set" capability is supported:
+        - The nexthop group is updated with the ARS object ID.
+    * If "set" capability is not supported - implement via "create" functionality:
+        - Create a new nexthop group with the ARS object ID.
+        - Redirect routes to use new nexthop group
+        - Remove original nexthop group.
+    * If an alternative path is defined
+        - Set the relevant nexthops as part of the alternative path.
+
+__Figure 7: Nexthop group creation flow__
+![](images/nhg_create.png)
+
+1. Nexthop group creation
+    * RouteOrch receives route from APPL_DB, triggering the process.
+    * Nexthop group is created
+2. Notification
+    * RouteOrch component will notify ArsOrch of NHG update.
+3. Nexthop group's SAI attributes handling:
+    * If nexthop group's ARS object ID, "set" capability is supported:
+        - The nexthop group is updated with the ARS object ID.
+    * If "set" capability is not supported - implement via "create" functionality:
+        - Create a new nexthop group with the ARS object ID.
+        - Redirect routes to use new nexthop group
+        - Remove original nexthop group.
+    * If an alternative path is defined
+        - Set the relevant nexthops as part of the alternative path.
+
+__Figure 8: Nexthop group update flow__
+![](images/nhg_update.png)
+
+1. Nexthop group update
+    * RouteOrch receives route update from APPL_DB, triggering the process.
+    * Nexthop group is updated with new member
+2. Notification
+    * RouteOrch component will notify ArsOrch of NHG update.
+4. Nexthop group's SAI attributes handling:
+    * If an alternative path is defined
+        - Set the relevant nexthops as part of the alternative path.
+
+__Figure 9:ARS LAG table creation flow__
+![](images/ars_lag_create.png)
+

```

Comment:
Please make sure that ARS on LAG is wrapped with the SAI capability check. Some vendors can support ARS for NHG but not for LAG